js/app.js

```diff
@@ -317,8 +320,38 @@ document.querySelector('#contact-form').addEventListener('submit', (e) => {
     }
   }
 
-  if (!valid) {
+  // if the email is valid, save user data in object
+  if (valid) {
+    // prevent the form fro automatically reloading the page
+    e.preventDefault();
+
+    // empty the response test in contact form
+    responseContainer.innerHTML = '';
+    // save the user info to object
+    const details = {
+      name,
+      email,
+      message,
+    };
+
+    // save the user object to localStorage
+    // (here we use JSON.stringify() to convert the user data to string for safe storage)
+    localStorage.setItem('contact-details', JSON.stringify(details));
```

Comment:
- [ ] Kindly save the data in local storage on every change in every input, so if the user enters some data and left the page when he came back he can see the data that he filled. You can use the `change` event instead of the `submit` event to make it.